### PR TITLE
Feature/hdinsight 3 5

### PR DESCRIPTION
**Pull request by dereklwood**
updates for HDInsight release using CDAP 3.5.0
- [x] update CDAP version
- [x] use `hdp-select` to determine HDP version (previous `ls` didn't handle multiple versions)
- [x] use future default router port `11015`
- [x] pin checkout to `v3.5.0` tag

### CODE REVIEW

File: cdap-distributions/src/hdinsight/install.sh [hunk 1]
```diff
@@ -78,7 +78,8 @@ source ${__gitdir}/cdap-common/bin/common.sh || die "Cannot source CDAP common s
 __zk_quorum=$(cdap_get_conf 'hbase.zookeeper.quorum' '/etc/hbase/conf/hbase-site.xml') || die "Cannot determine zookeeper quorum"
 
 # Get HDP version, allow for the future addition hdp-select "current" directory
-__hdp_version=$(ls /usr/hdp | grep "^[0-9]*\.") || die "Cannot determine HDP version"
+__hdp_version_str=$(hdp-select status hadoop-client) || die "Cannot run hdp-select to determine HDP version"
+__hdp_version=$(echo ${__hdp_version_str} | cut -d' ' -f3) || die "Cannot determine HDP version from string ${__hdp_version_str}"
```
Comment (wolf31o2) [inline cdap-distributions/src/hdinsight/install.sh:82]: Why not pipe and cut in one go?

Comment (dereklwood) [reply cdap-distributions/src/hdinsight/install.sh:82]: I didn't want the cut to mask any errors from hdp-select.  maybe overkill but cut seems to never return non-zero

Comment (wolf31o2) [reply cdap-distributions/src/hdinsight/install.sh:82]: Fair enough... may want to comment as much

**Comment by wolf31o2** [comment]
👍 LGTM

**Pull request merged**